src/wolfsftp.c

```diff
@@ -3561,19 +3648,140 @@ int SFTP_GetAttributes(const char* fileName, WS_SFTP_FILEATRB* atr, byte link,
     return WS_SUCCESS;
 }
 
+#elif defined(FREESCALE_MQX)
+
+/* @TODO can be overriden by user for portability
+ * NOTE: if atr->flags is set to a value of 0 then no attributes are set.
+ * Fills out a WS_SFTP_FILEATRB structure
+ * returns WS_SUCCESS on success */
+int SFTP_GetAttributes(void* fs, const char* fileName, WS_SFTP_FILEATRB* atr,
+                       byte link, void* heap)
+{
+    int err, sz;
+    MQX_FILE_PTR mfs_ptr;
+    MFS_SEARCH_DATA search_data;
+    MFS_SEARCH_PARAM search;
+
+    if (fileName == NULL || atr == NULL) {
+        return WS_BAD_FILE_E;
+    }
+    mfs_ptr = (MQX_FILE_PTR)fs;
+    sz = (int)WSTRLEN(fileName);
```

Comment:
check that sz is at least 3 or greater for when doing the WSTRNCMP later (fileName + sz - 3)

Reply:
Good catch, will change.

Reply:
Fixed